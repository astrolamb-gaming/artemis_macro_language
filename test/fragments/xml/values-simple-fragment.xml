<mission_data version="1.0">
  <values>
    <value CONST_TIMER_SECONDS="30"/>
    <struct name="mystructure" >
      <value anomaly="Anomaly[1]" angle="90" pickupType="5" object="A1" y="1.0" />
      <struct name="inner" >
        <value health="5" />
        <value XP="34" />
      </struct>   
    </struct>
    <range name="test">
      <struct name="inner" >
        <value health="6" />
        <value XP="34" />
      </struct>
    </range>
  </values>

  <start>
    <expand>
      <set_variable name="${mystructure.anomaly}" value="${CONST_TIMER_SECONDS*5}"/>
      <set_variable name="${mystructure.anomaly}" value="${CONST_TIMER_SECONDS}"/>
      <set_variable name="health" value="${mystructure.inner.health}"/>
    </expand>
    <repeat range="test">
      <set_variable name="health" value="${health}"/>
    </repeat>
  </start>

  
</mission_data>