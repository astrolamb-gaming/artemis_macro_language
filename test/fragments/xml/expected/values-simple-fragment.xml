<mission_data version="1.0">
  <start>
    <set_variable name="Anomaly[1]" value="150"/>
    <set_variable name="Anomaly[1]" value="30"/>
    <set_variable name="health" value="5"/>
    <set_variable name="health" value="6"/>
  </start>
</mission_data>